tests/cupy_tests/core_tests/fusion_tests/test_function.py

```diff
@@ -0,0 +1,540 @@
+import threading
+import unittest
+
+import mock
+import six
+
+import cupy
+from cupy import testing
+from cupy_tests.core_tests.fusion_tests import fusion_utils
+
+
+class FusionTestBase(unittest.TestCase):
+    def generate_inputs(self, xp, nargs, dtype):
+        inputs = [
+            testing.shaped_random((3, 4), xp, dtype, scale=10, seed=seed)
+            for seed in six.moves.range(nargs)
+        ]
+        return inputs, {}
+
+    def dtype_combination(self, xp, dtype1, dtype2):
+        x = testing.shaped_random((3, 4), xp, dtype1, scale=10, seed=0)
+        y = testing.shaped_random((3, 4), xp, dtype2, scale=10, seed=1)
+        return (x, y), {}
+
+
+@testing.gpu
+class TestFusionOutArg(unittest.TestCase):
+
+    def generate_inputs(self, xp, dtype1, dtype2):
+        x = testing.shaped_random((3, 4), xp, dtype1, scale=10, seed=0)
+        y = testing.shaped_random((3, 4), xp, dtype1, scale=10, seed=1)
+        z = testing.shaped_random((3, 4), xp, dtype2, scale=10, seed=2)
+        return (x, y, z), {}
+
+    @testing.for_all_dtypes_combination(
+        names=('dtype1', 'dtype2'), full=True, no_complex=True)
+    @fusion_utils.check_fusion(accept_error=TypeError)
+    @testing.with_requires('numpy>=1.13')
```

Comment:
Note: Casting rule have changed from NumPy 1.13 ([Release notes](https://docs.scipy.org/doc/numpy-1.13.0/release.html#default-casting-rule-change))

Reply:
Why hasn't this been a problem before this PR?